src/tests/src/libs/antares/study/short-term-storage-input/short-term-storage-input.cpp

```diff
@@ -0,0 +1,279 @@
+    outfile << "reservoircapacity = 31200.000000" << std::endl;
+    outfile << "efficiency = 0.75" << std::endl;
+    outfile << "storagecycle = 50" << std::endl;
+    outfile << "initiallevel = 0.50000" << std::endl;
+
+    outfile.close();
+}
+
+void createIniFileWrongValue()
+{
+    std::string folder = getFolder();
+
+    std::ofstream outfile;
+    outfile.open(folder + SEP + "list.ini", std::ofstream::out | std::ofstream::trunc);
+
+    outfile << "[area]" << std::endl;
+    outfile << "name = area" << std::endl;
+    outfile << "group = abcde" << std::endl;
+    outfile << "injectionnominalcapacity = -870.000000" << std::endl;
+    outfile << "withdrawalnominalcapacity = -900.000000" << std::endl;
+    outfile << "reservoircapacity = -31200.000000" << std::endl;
+    outfile << "efficiency = 4" << std::endl;
+    outfile << "storagecycle = 50" << std::endl;
+    outfile << "initiallevel = -0.50000" << std::endl;
+
+    outfile.close();
+}
+
+void createEmptyIniFile()
+{
+    std::string folder = getFolder();
+
+    std::ofstream outfile;
+    outfile.open(folder + SEP + "list.ini", std::ofstream::out | std::ofstream::trunc);
+
+    outfile.close();
+}
+
+void removeIniFile()
+{
+    std::string folder = getFolder();
+    std::filesystem::remove(folder + SEP + "list.ini");
+}
+
+// =================
+// The fixture
+// =================
+struct Fixture
+{
+    Fixture(const Fixture & f) = delete;
+    Fixture(const Fixture && f) = delete;
+    Fixture & operator= (const Fixture & f) = delete;
+    Fixture& operator= (const Fixture && f) = delete;
+    Fixture()
+    {
+        cluster.series = series;
+        cluster.properties = properties;
+    }
+
+    ~Fixture() = default;
+    ShortTermStorage::Series series;
+    ShortTermStorage::Properties properties;
+    ShortTermStorage::STStorageCluster cluster;
+    ShortTermStorage::STStorageInput container;
+
+};
+
+
+// ==================
+// Tests section
+// ==================
+
+BOOST_FIXTURE_TEST_SUITE(s, Fixture)
+
+BOOST_AUTO_TEST_CASE(check_vector_sizes)
+{
+    resizeFillVectors(series, 0.0, 12);
+    BOOST_CHECK(!series.validate());
+
+    resizeFillVectors(series, 0.0, 8760);
+    BOOST_CHECK(series.validate());
+}
+
+BOOST_AUTO_TEST_CASE(check_series_folder_loading)
+{
+    std::string folder = getFolder();
+
+    createFileSeries(1.0, 8760);
+
+    BOOST_CHECK(series.loadFromFolder(folder));
+    BOOST_CHECK(series.validate());
+
+    removeFileSeries();
+}
+
+BOOST_AUTO_TEST_CASE(check_series_folder_loading_negative_value)
+{
+    std::string folder = getFolder();
+
+    createFileSeries(-247.0, 8760);
+
+    BOOST_CHECK(series.loadFromFolder(folder));
+    BOOST_CHECK(series.validate());
+
+    removeFileSeries();
+}
+
+BOOST_AUTO_TEST_CASE(check_series_folder_loading_too_big)
+{
+    std::string folder = getFolder();
+
+    createFileSeries(1.0, 9000);
+
+    BOOST_CHECK(series.loadFromFolder(folder));
+    BOOST_CHECK(series.validate());
```

Comment:
This shouldn't pass, right ?

Reply:
We give the size to the function that loads the array, it reads just what is needed to it pass